File: oo_modular_design/src/CarpetQuote.java
class CarpetQuote {

    double quote(double width, double length, double pricePerSqrMtr, boolean roundUp){
        double area = width * length;

        if(roundUp) {
            area = Math.ceil(area);
        }

        return pricePerSqrMtr * area;
    }
}

File: modular_design/src/CarpetQuote.kt
fun quote(room: Pair<Double, Double>, carpet: Pair<Double, Boolean>) : Double {
    val (width, length) = room
    val (pricePerSqrMtr, roundUp) = carpet

    var area = width * length

    if(roundUp){
        area = Math.ceil(area)
    }

    return pricePerSqrMtr * area
}